### PR TITLE
[Concept Entry] Git Restore

### LINKED ISSUE #3493: [Concept Entry] Git Restore
### Reason/inspiration (optional)

We would like a new concept entry for Git about Restore. The entry should go in a new file in docs/content/git/concepts/restore/restore.md

The entry should
    - Have an introductory paragraph about the concept.
    - Code blocks displaying different aspects of the concept.
    - Internal linking as applicable.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by krushnarout**
### Description

Added new concept entry for Git about Restore.

Closes #3493 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/git/concepts/restore/restore.md
Comment (CBID2) [inline content/git/concepts/restore/restore.md:23]: Since this focuses more on the syntax, it would be best to describe it as `pseudo` code. 


```suggestion
```pseudo
git restore <options> -- <file>
```
```

File: content/git/concepts/restore/restore.md
Comment (CBID2) [inline content/git/concepts/restore/restore.md:26]: [suggestion] Consider putting a code snippet that often uses the `restore` command. That way developers and other people who use coding can see how it is used.  
```suggestion
add new code snippet here
```

File: content/git/concepts/restore/restore.md
Comment (CBID2) [inline content/git/concepts/restore/restore.md:27]: [suggestion] From there, you can modify the description based on your new code snippet
```suggestion
In this example, the initial [`git status`](https://www.codecademy.com/resources/docs/git/status) command shows that there are changes in the `index.html` file that have not been staged for a commit. It suggests using [`git add`](https://www.codecademy.com/resources/docs/git/add) to stage these changes. However, the user decides to use `git restore index.html` to discard the changes in the working directory and revert the file to its last committed state. After running this command and checking [`git status`](https://www.codecademy.com/resources/docs/git/status) again, the working directory is now clean, indicating that there are no pending changes, and there's nothing left to commit.
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @krushnarout. Your PR is off to a great start. :) I gave some tips on how you can improve your entry. If you need more inspiration, check out my [entry with the `cherry pick `command](https://www.codecademy.com/resources/docs/git/cherry-pick)

File: content/git/concepts/restore/restore.md
Comment (krushnarout) [reply content/git/concepts/restore/restore.md:26]: Hi! Thank you for your suggestion. I believe that I have added a `git restore`code snippet example below the explanation. Could you please explain it to me in more detail?

**Comment by krushnarout** [comment]
@CBID2 Hi! Thank you for your suggestion. I believe that I have added a `git restore` code snippet example below the explanation. Could you please explain it to me in more detail?

File: content/git/concepts/restore/restore.md
Comment (CBID2) [reply content/git/concepts/restore/restore.md:26]: Hi @krushnarout, I modified my comment. Check it out

File: content/git/concepts/restore/restore.md
Comment (CBID2) [inline content/git/concepts/restore/restore.md:27]:
```suggestion
In the above example, the initial [`git status`](https://www.codecademy.com/resources/docs/git/status) command shows that there are changes in the `index.html` file that have not been staged for a commit. It suggests using [`git add`](https://www.codecademy.com/resources/docs/git/add) to stage these changes. However, the user decides to use `git restore index.html` to discard the changes in the working directory and revert the file to its last committed state. After running this command and checking [`git status`](https://www.codecademy.com/resources/docs/git/status) again, the working directory is now clean, indicating that there are no pending changes, and there's nothing left to commit.
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Made a change

**Comment by krushnarout** [comment]
@CBID2 Thank you

**Comment by avdhoottt** [comment]
Hey @CBID2 I'm glad that you showed your interest in reviewing the PRs but unfortunately, only the internal team (people with the collaborator tag) can review the PRs and give feedback. But you can check out the [open issues](https://github.com/Codecademy/docs/issues) and ask the maintainers to assign it to you. Thank you for collaborating!!

**Comment by CBID2** [comment]
> Hey @CBID2 I'm glad that you showed your interest in reviewing the PRs but unfortunately, only the internal team (people with the collaborator tag) can review the PRs and give feedback. But you can check out the [open issues](https://github.com/Codecademy/docs/issues) and ask the maintainers to assign it to you. Thank you for collaborating!!

Oh I didn't know that @avdhoottt. Is this a new rule? Usually, my reviews are accepted.

**Comment by avdhoottt** [comment]
> > Hey @CBID2 I'm glad that you showed your interest in reviewing the PRs but unfortunately, only the internal team (people with the collaborator tag) can review the PRs and give feedback. But you can check out the [open issues](https://github.com/Codecademy/docs/issues) and ask the maintainers to assign it to you. Thank you for collaborating!!
> 
> Oh I didn't know that @avdhoottt. Is this a new rule? Usually, my reviews are accepted.

As of my knowledge Yes.

File: content/git/concepts/restore/restore.md
Comment (avdhoottt) [inline content/git/concepts/restore/restore.md:15]:
```suggestion
The **`restore`** command enables developers to restore files to a specific state. This functionality allows you to undo any changes made to files in both your working directory and staging area. This command is especially valuable when you need to revert alterations, whether they have been staged for a commit or not.
```

File: content/git/concepts/restore/restore.md
Comment (avdhoottt) [inline content/git/concepts/restore/restore.md:58]: Move this below the syntax of the Restore command.

Comment (avdhoottt) [inline content/git/concepts/restore/restore.md:58]:
```suggestion
- `-s <tree> --source=<tree>`: This option allows you to restore the working tree files with the content from the given tree.
- `-p --patch`: This option allows you to interactively select hunks in the difference between the restore source and the restore location.
- `-W --worktree -S --staged`: These options specify the restore location. If neither option is specified, by default the working tree is restored.
- `-q --quiet`: This option suppresses feedback messages and implies `--no-progress`.
- `--progress --no-progress`: These options control progress status reporting.
- `--ours --theirs`: These options are used when restoring files in the working tree from the index, to use stage #2 (ours) or #3 (theirs) for unmerged paths.
- `-m --merge`: This option recreates the conflicted merge in the unmerged paths when restoring files on the working tree from the index.
- `--conflict=<style>`: This option changes the way conflicting hunks are presented, overriding the `merge.conflictStyle` configuration variable.
- `--ignore-unmerged`: This option allows you to restore files on the working tree from the index without aborting if there are unmerged entries.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@krushnarout Some changes from my side.

**Comment by krushnarout** [comment]
@avdhoottt I've updated the entry with the suggested changes. Please review and let me know if any further changes are needed. Thanks!

**Review by avdhoottt — APPROVED** [review]
Ready for the second review.

File: content/git/concepts/restore/restore.md
Comment (dakshdeepHERE) [inline content/git/concepts/restore/restore.md:15]:
```suggestion
The **`restore`** command enables developers to restore files to a specific state. This functionality allows restoring any changes made to files in both the working directory and staging area. This command is particularly valuable when reverting alterations, whether staged for a commit or not.
```
We don't prefer using pro nouns

File: content/git/concepts/restore/restore.md
Comment (dakshdeepHERE) [inline content/git/concepts/restore/restore.md:35]:
```suggestion
Nine additional options are available for `git restore`:

- `-s <tree> --source=<tree>`: Allows restoring the working tree files with content from the specified tree.
- `-p --patch`: Enables interactive selection of hunks between the restore source and location.
- `-W --worktree -S --staged`: Specifies the restore location. If neither is specified, the working tree is restored by default.
- `-q --quiet`: Suppresses feedback messages and implies `--no-progress`.
- `--progress --no-progress`: Controls progress status reporting.
- `--ours --theirs`: Used when restoring files in the working tree from the index, to use stage #2 (ours) or #3 (theirs) for unmerged paths.
- `-m --merge`: Recreates the conflicted merge in unmerged paths when restoring files to the working tree from the index.
- `--conflict=<style>`: Modifies the presentation of conflicting hunks, overriding the `merge.conflictStyle` configuration variable.
- `--ignore-unmerged`: Allows restoring files to the working tree from the index without aborting due to unmerged entries.
```
refrain from using pronouns

File: content/git/concepts/restore/restore.md
Comment (dakshdeepHERE) [inline content/git/concepts/restore/restore.md:39]: You have mentioned here `In the above example` which above example you are referring to here?

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have few comments for you to work on. :)

**Comment by krushnarout** [comment]
@dakshdeepHERE I've updated the entry with the suggested changes. Please review and let me know if any further changes are needed. Thanks!

**Comment by dakshdeepHERE** [comment]
Hey @krushnarout let me look through again and see if any changes is required. If not we will merge.

**Comment by dakshdeepHERE** [comment]
@krushnarout Thanks for this entry it looks good to merge! Looking forward to see more from your side.🚀

**Review by dakshdeepHERE — APPROVED** [review]

**Pull request merged**
